File: back/src/main/java/com/csid/justread/book/exposition/dto/BookDto.java
package com.csid.justread.book.exposition.dto;

import com.csid.justread.book.infrastructure.entity.PublisherEntity;

import javax.persistence.Column;
import javax.persistence.FetchType;
import javax.persistence.ManyToOne;
import java.util.List;
import java.util.UUID;

public class BookDto {

    public BookDto() {}

    private UUID id;
    private String title;
    private String synopsys;
    private AuthorDto author;
    private List<CategoryDto> categories;
    private String isbn;
    private Integer year;
    private PublisherDto publisher;

    //region * Methods *

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getSynopsys() {
        return synopsys;
    }

    public void setSynopsys(String synopsys) {
        this.synopsys = synopsys;
    }

    public AuthorDto getAuthor() {
        return author;
    }

    public void setAuthor(AuthorDto author) {
        this.author = author;
    }

    public List<CategoryDto> getCategories() {
        return categories;
    }

    public void setCategories(List<CategoryDto> categories) {
        this.categories = categories;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }

    public PublisherDto getPublisher() {
        return publisher;
    }

    public void setPublisher(PublisherDto publisher) {
        this.publisher = publisher;
    }

    //endregion

}

File: front/src/app/core/Dtos/BookDto.ts
export interface AuthorDto {
      id: string;
      firstname: string;
      lastname: string;
      pseudo: string;
      urlWiki: string;
      bookCount: number;
}
export interface CategoryDto {
    name: string;
}
export interface PublisherDto {
    //TODO
}

export interface stockDto{
    id: string | undefined;
    book: BookDto;
    quantity: number;
    unitPrice: number;
}

export interface BookDto {

    id: string;
    title: string;
    synopsys: string;
    author?: AuthorDto | null;
    categories: CategoryDto[];
    isbn: string;
    year: number;
    urlImage: string;
    publisher: PublisherDto;

}